receiver/receivertest/contract_checker.go

```diff
@@ -0,0 +1,405 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package receivertest // import "go.opentelemetry.io/collector/receiver/receivertest"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.opentelemetry.io/collector/receiver"
+)
+
+// UniqueIDAttrName is the attribute name that is used in log records/spans/datapoints as the unique identifier.
+const UniqueIDAttrName = "test_id"
+
+// UniqueIDAttrDataType is the value type of the UniqueIDAttrName.
+type UniqueIDAttrDataType int64
+
+type Generator interface {
+	// Generate must generate and send at least one data element (span, log record or metric data point)
+	// to the receiver and return a copy of generated element ids.
+	// The generated data must contain uniquely identifiable elements, each with a
+	// different value of attribute named UniqueIDAttrName.
+	// CreateOneLogWithID() can be used a helper to create such logs.
+	Generate() IDSet
```

Comment:
If we change this signature to `Generate() []UniqueIDAttrDataType`, we don't need to expose the `IDSet` anymore, which seems not very useful to the end user at this point. WDYT?